terra_node_bot.py

```diff
@@ -0,0 +1,651 @@
+        if response.status_code == 200:
+            return response.json()
+        else:
+            return None
+
+
+def get_node_status_endpoint():
+    """
+    Return the endpoint for block height checks
+    """
+
+    return 'http://localhost:8000/status.json' if DEBUG else 'http://' + NODE_IP + ':26657/status'
+
+
+"""
+######################################################################################################################################################
+Jobs
+######################################################################################################################################################
+"""
+
+
+def node_checks(context):
+    """
+    Periodic checks of various node stats
+    """
+
+    check_node_status(context)
+    check_price_feeder(context)
+    if NODE_IP:
+        check_node_catch_up_status(context)
+        check_node_block_height(context)
+
+
+def check_node_status(context):
+    """
+    Check all added Terra Nodes for any changes.
+    """
+
+    chat_id = context.job.context['chat_id']
+    user_data = context.job.context['user_data']
+
+    # Flag to show home buttons or not
+    message_sent = False
+
+    # List to delete entries after loop
+    delete_addresses = []
+
+    # Iterate through all keys
+    for address in user_data['nodes'].keys():
+        remote_node = get_validator(address=address)
+        local_node = user_data['nodes'][address]
+
+        if remote_node is None:
+            text = 'Node is not active anymore! 💀' + '\n' + \
+                   'Address: ' + address + '\n\n' + \
+                   'Please enter another Node address.'
+
+            delete_addresses.append(address)
+
+            # Send message
+            context.bot.send_message(chat_id, text)
+            message_sent = True
+            continue
+
+        # Check which node fields have changed
+        changed_fields = [field for field in ['status', 'jailed'] if
+                          local_node[field] != remote_node[field]]
+
+        # Check if there are any changes
+        if len(changed_fields) > 0:
+            text = 'Node: ' + address + '\n' + \
+                   'Status: ' + NODE_STATUSES[local_node['status']]
+            if 'status' in changed_fields:
+                text += ' ➡️ ' + NODE_STATUSES[remote_node['status']]
+            text += '\nJailed: ' + str(local_node['jailed'])
+            if 'jailed' in changed_fields:
+                text += ' ➡️ ' + str(remote_node['jailed'])
+
+            # Update data
+            local_node['status'] = remote_node['status']
+            local_node['jailed'] = remote_node['jailed']
+
+            # Send message
+            context.bot.send_message(chat_id, text)
+            message_sent = True
+
+    for address in delete_addresses:
+        del user_data['nodes'][address]
+
+    if message_sent:
+        show_home_menu_new_msg(context=context, chat_id=chat_id)
+
+
+def check_price_feeder(context):
```

Comment:
Should we also do price validity checks? So if the commited prices are within a certain %-range?